### PR TITLE
Classic Theme Helper: Copy Social Links code to Classic Theme Helper package 

### PR DESCRIPTION
**Pull request by coder-karen**
Fixes https://github.com/Automattic/vulcan/issues/431

## Proposed changes:

* Moved the main file `social-links.php`.
* In order to avoid any issues if the package file happens to load just before the module file (as the package file is being read in my local tests), I've commented out the class initialization from the package file for now, and will modify this when requiring the package in the next PR.
* Added some Phan suppressions and attempted some Phan suggested fixes related to adding class / function checks, but these won't be tested until requiring this feature via the package (in the next PR).
* There is no WebPack build added, as there is none in the Jetpack plugin either for this feature.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [x] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [x] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

https://github.com/Automattic/vulcan/issues/431

## Does this pull request change what data or activity we track or use?

No.

## Testing instructions:

* Technically the package isn't being used yet, and no build is happening, so a code check is the main thing.
* However in testing I did come across a fatal and had to make a few changes to make sure the package version didn't try to initialize the class as the file was being read, so we should make sure functionality is not affected. To do so:
  
Self-hosted and Atomic:
* On a test site, add a theme that includes support for Jetpack's social-links. The only theme I could find is 'Syntax'. 
* Otherwise, make sure you have a classic (not block) theme, such as Twenty Nineteen, and add the following snippet with a functionality plugin (like 'Code Snippets'):
```
add_action( 'after_setup_theme', function () {
	add_theme_support( 'social-links', array(
		'facebook', 'twitter', 'linkedin', 'tumblr',
	) );
} );
```
* Enable Social sharing - `/wp-admin/admin.php?page=jetpack#/sharing` and add a connection. 
* You can add a Facebook page by using the Gmail username and pw for `credsflow....` in the SS (internal only).
* Visit the Customizer, and you should see a 'Connect' menu item, that allows you to select your newly added connection.
* This should work both on trunk and with this PR applied.
* On Atomic make sure to test with the PR checked out for both Jetpack and WPcom Features, and each individually.

Simple:
* Add a social connection at `https://wordpress.com/marketing/connections/yoursite.com`
* Add a theme which includes support for Jetpack's social-links - Syntax for example.
* Visit the Customizer and you should see the 'Connect' menu item allowing you to connect your new social connection.
* This should work both with and without the patches applied - see testing comment below (and also one at a time to replicate Jetpack only and Mu Wpcom only).

### CODE REVIEW

File: projects/packages/classic-theme-helper/src/social-links.php [hunk 1]
```diff
@@ -0,0 +1,279 @@
...
+				return;
+			}
+
+			if ( ! is_admin() && ! $this->is_customize_preview() ) {
+				return;
+			}
+
+			// @phan-suppress-next-line PhanUndeclaredFunction -- Function checked with function_exists - see https://github.com/phan/phan/issues/1204.
+			$this->publicize    = function_exists( 'publicize_init' ) ? publicize_init() : null;
+			$publicize_services = $this->publicize->get_services( 'connected' );
```
Comment (darssen) [inline projects/packages/classic-theme-helper/src/social-links.php:113]: What happens here if `$this->publicize` is null?

**Review by darssen — COMMENTED** [review]
Thanks for working on this one @coder-karen!

Changes look sensible to me before testing.

As of now, there is only two things that I can think of that might need changes.
1. I'm not sure how much sense does all of the logic if `Publicize`  or `publicize_init` don't exist. I see there is a wrapper inside `customize_register(` checking it, but I feel like maybe it would need to be early on. Should it be even before that? Maybe return early in the constructor? I have added an inline comment in a place where I think the absence of Publicize will create a Fatal
2. I would change is the filename `projects/packages/classic-theme-helper/src/social-links.php` to `projects/packages/classic-theme-helper/src/class-social-links.php`.

File: projects/packages/classic-theme-helper/src/social-links.php (hunk 1 above)
Comment (coder-karen) [reply projects/packages/classic-theme-helper/src/social-links.php:113]: Good question. I've no idea at this stage, but I'll make sure to look into that when requiring the package in the next PR.

**Comment by coder-karen** [comment]
Thanks for testing @darssen !

> I'm not sure how much sense does all of the logic if Publicize or publicize_init don't exist. 

Good questions, thanks! What I'll do is make sure to investigate that when requiring the package in the next PR. None of that moved functionality should be running here, only the existing functionality.

I'll add a note to the issue for the next step: https://github.com/Automattic/vulcan/issues/432

> I would change is the filename projects/packages/classic-theme-helper/src/social-links.php to projects/packages/classic-theme-helper/src/class-social-links.php.

Yes, I did initially try that, and it resulted in some issues for me (a fatal error). I had added that as something to try and rename in the next PR - added in the above issue. The issue seemed related to the Jetpack autoloader. Had you come across this before? I know Featured Content was also in a class so I was looking at what was done there originally.

**Review by monsieur-z — APPROVED** [review]
This tested well with all types of sites.

**Pull request merged**